repository/instance_repository.go

```diff
@@ -0,0 +1,71 @@
+package repository
+
+import (
+	"errors"
+	"fmt"
+
+	logKeys "github.com/ONSdigital/dp-dimension-importer/common"
+	"github.com/ONSdigital/dp-dimension-importer/model"
+	"github.com/ONSdigital/go-ns/log"
+)
+
+const (
+	// Create an Insatnce node.
+	createInstanceStmt = "CREATE (i:`%s`) RETURN i"
+
+	// Update the Instance node with the list of dimension types it contains.
+	addInstanceDimensionsStmt = "MATCH (i:`%s`) SET i.dimensions = {dimensions_list}"
+
+	instanceIDReqErr             = "Instance.InstanceID is required but was empty"
+	createInstanceExecErr        = "Error while executing to create Instance statement."
+	createInstanceSuccess        = "Successfully created Instance node."
+	addInstanceDimensionsExecErr = "Error while executing add instance dimensions statement"
+	addInstanceDimensionsSuccess = "Successfully added dimensions to Instance node"
+)
+
+type InstanceRepository struct {
+	Neo NeoDatabase
+}
+
+func (repo *InstanceRepository) Create(i *model.Instance) error {
+	if len(i.InstanceID) == 0 {
+		err := errors.New(instanceIDReqErr)
+		log.ErrorC(instanceIDReqErr, err, nil)
+		return err
+	}
+
+	stmt := fmt.Sprintf(createInstanceStmt, i.GetLabel())
+	logDebug := map[string]interface{}{
+		logKeys.InstanceID: i.InstanceID,
+		stmtKey:            stmt,
+	}
+
+	if _, err := repo.Neo.ExecStmt(stmt, nil); err != nil {
```

Comment:
Adding a note for this to the tech dept wall - This needs to be applied to the observation importer also.